corehq/apps/es/filters.py

```diff
@@ -38,31 +38,17 @@ def term(field, value):
 
 def OR(*filters):
     """Filter docs to match any of the filters passed in"""
-    return {"or": filters}
+    return {"bool": {"should": filters}}
 
 
 def AND(*filters):
     """Filter docs to match all of the filters passed in"""
-    return {"and": filters}
+    return {"bool": {"filter": filters}}
 
 
 def NOT(filter_):
     """Exclude docs matching the filter passed in"""
-    if 'or' in filter_:
-        # ES 2.4 appears not to accept {"not": {"or": [A, B]}} e.g. not (A or B)
-        # but accepts the same logic
-        # formulated as {"and": [{"not": A}, {"not": B}]} (e.g. not A and not B)
-        return AND(*(NOT(condition) for condition in filter_['or']))
-    elif 'and' in filter_:
-        # Same ES 2.4 issue as above.
-        # Rewrite "not (A and B)" as "not A or not B"
-        return OR(*(NOT(condition) for condition in filter_['and']))
-    elif 'not' in filter_:
-        # This may not be strictly necessary
-        # but prevents {'not': {'not': A}}, in favor of just A
-        return filter_['not']
-    else:
-        return {"not": filter_}
+    return {"bool": {"must_not": filter_}}
```

Comment:
I added a test case to make sure this works without the rewrite. See https://github.com/dimagi/commcare-hq/pull/28372/commits/89c1a8d19820699ccf08a9139cd972193c265b91

Reply:
FYI @dannyroberts @esoergel

Reply:
Ah, I wonder if the issue Danny identified was only present when using `{"and": {}}` and `{"or": {}}`, and not with the `bool` implementations.

Reply:
May be.